calculator.go

```diff
@@ -30,3 +80,73 @@ func Divide(a, b float64) (float64, error) {
 	}
 	return a / b, nil
 }
+
+// DivideN takes a slice of numbers and returns the division of the given numbers
+// ((num1 / num2) / num3) / num4 etc..
+func DivideN(numbers []float64) (float64, error) {
+	length := len(numbers)
+
+	if length == 0 {
+		return 0, nil
+	}
+	if length == 1 {
+		return numbers[0], nil
+	}
+
+	result := numbers[0]
+	for i := 1; i < length; i++ {
+		if numbers[i] == 0 {
+			return 0, errors.New("Divide by 0")
+		}
+		result /= numbers[i]
+	}
+	return result, nil
+}
+
+// SquareRoot returns the square root of the provided number , error if the input
+// number is invalid
+func SquareRoot(a float64) (float64, error) {
+	if a < 0 {
+		return 0, errors.New("negative numbers don't have square root")
+	}
+	return math.Sqrt(a), nil
+}
+
+// ComputeString processes the input string as two operands separated by the operator
+// and returns the result of that operation. Returns error if the string is not in
+// the acceptable form.
+func ComputeString(str string) (float64, error) {
+	for _, op := range "+-*/" {
```

Comment:
It might be clearer to try to parse the string into 'A, operator, B' first, then switch on the operator to compute the result.